src/main/java/mate/academy/MergeSortAction.java

```diff
@@ -1,8 +1,35 @@
 package mate.academy;
 
-public class MergeSortAction {
+import java.util.Arrays;
+import java.util.concurrent.RecursiveAction;
+
+public class MergeSortAction extends RecursiveAction {
+
+    private static final int THRESHOLD = 3;
+    private final int[] array;
+    private final int length;
+    private final int start;
 
     public MergeSortAction(int[] array) {
+        this(array, 0, array.length);
+    }
+
+    private MergeSortAction(int[] array, int length, int start) {
+        this.array = array;
+        this.length = length;
+        this.start = start;
```

Comment:
The parameter names for the private constructor seem to be swapped. The 'length' parameter should represent the end index of the subarray to sort, and 'start' should represent the start index. As it stands, the names are misleading and do not reflect their actual use in the code.